### DISCUSSION TITLE
Callstel Tracker 4in1

### DISCUSSION THREAD

**Discussion by gnome4711** [ESPresense/ESPresense#484] [General]
Hi there!

This for general information for those who are thinking about buying a Callstel Tracker4in1 (https://www.amazon.de/gp/product/B0979DYV53)

Using vers. 2.1.1:

I own 2 Callstel Tracker 4in1. Unfortunately ESPresense creates the same id for the both of them:  ID: name:itrack
It is not a big deal, but I have to add the published mac-address (mqtt) to the topic or to the id. After that workaround it can be used as any other beacon. But I think ESPresense should make an unique id by itself.

Maybe in the beta versions it is already fixed.

All previous version I tested so far didn't recognize the Callstel Tracker at all.

**Comment by DTTerastar** [comment]
I'd need a scan via nRF Connect to find something unique about it.

**Reply by gnome4711** [reply]
[DTTerastar](https://github.com/DTTerastar)

Here is the output of nRF Connect. I made some screenshots (phone). It is only one of the 2 iTracks I use. If you need a scan of the other one, I have to wait until my daughter is home. She has it in her wallet. 

![Screenshot_2022-06-06-14-39-30-79_3c2e989be58abe9ca7463ffb1e6967d2](https://user-images.githubusercontent.com/7687563/172338158-519d54a1-1cce-4743-9649-dde9ddba44fe.jpg)
![Screenshot_2022-06-06-14-39-03-54_3c2e989be58abe9ca7463ffb1e6967d2](https://user-images.githubusercontent.com/7687563/172338162-32e786d3-62cf-4d87-91c3-d8ef90002ac2.jpg)
![Screenshot_2022-06-06-14-59-30-94_3c2e989be58abe9ca7463ffb1e6967d2](https://user-images.githubusercontent.com/7687563/172338165-fe3f07ce-88b3-481d-a1c4-afb56a42c8be.jpg)
![Screenshot_2022-06-06-14-39-39-36_3c2e989be58abe9ca7463ffb1e6967d2](https://user-images.githubusercontent.com/7687563/172338166-7456c7ae-c92f-4e8f-996f-11476726b952.jpg)

**Reply by DTTerastar** [reply]
That'll do, i'll key off of the manufacturer id

**Reply by DTTerastar** [reply]
It's in the latest beta, please confirm it works for you!

**Comment by gnome4711** [comment]
[DTTerastar](https://github.com/DTTerastar)

Still no unique id in latest beta!

![grafik](https://user-images.githubusercontent.com/7687563/174593835-ba54f86c-c367-4f1d-9cc8-e1ecfef73261.png)

**Reply by DTTerastar** [reply]
Hmmmmm, let me check my work...

**Reply by DTTerastar** [reply]
Yup, I did it wrong.  Next beta will have the fix.

**Reply by gnome4711** [reply]
[DTTerastar ](https://github.com/DTTerastar)

Yes! It works now. Thanks!!

[08:54:44]1 New   | MAC: f0bf24300000, ID: iTrack:f0bf24300000

**Reply by DTTerastar** [reply]
Awesome!
